Library/Homebrew/formulary.rb

```diff
@@ -200,8 +200,28 @@ def self.factory(ref, spec=:stable)
     loader_for(ref).get_formula(spec)
   end
 
+  # Return a Formula instance for the given rack.
+  def self.factory_from_rack(rack, spec=:stable)
```

Comment:
Should this be `formula_from_rack` (or just `from_rack`)? `factory_from_rack` sounds a little weird to me (`factory` itself is also probably a bad name).